### PR TITLE
OrgUnitProvisioning.findOrCreate caches and returns orgUnit json

### PR DESCRIPTION
**Pull request by kshen0**
`OrgUnitProvisioning.findOrCreate` previously returned an object mapping OU name to a boolean.

This PR updates `findOrCreate` to return a mapping of OU name to an object representation of the OU eg.

```
{
  "/MyOrgUnit": {
    kind: 'admin#directory#orgUnit',
    etag: '"LDSLG32KJLDJsdkfjSDfhf23kljsdf"',
    name: 'MyOrgUnit',
    orgUnitPath: '/MyOrgUnit',
    orgUnitId: 'id:56789dcba',
    parentOrgUnitPath: '/',
    parentOrgUnitId: 'id:12345abcde'
  }
}
```

### CODE REVIEW

File: lib/org_unit_provisioning.coffee [hunk 1]
```diff
@@ -88,19 +88,41 @@ class OrgUnit extends GoogleAPIAdminSDK
...
+            if err?
+              if err.error?.code is 404 and err.error?.message is "Org unit not found"
+                # Not found; pass through to create the orgunit
+                return cb_wf null, null
+              else
+                # A valid error was returned. Return the error and skip the insert.
+                return cb_wf err
+
+            # If no error, we found the orgunit. Cache it and skip insert because it already exists.
+            cache[full_path] = body
```
Comment (alexandersoto) [inline lib/org_unit_provisioning.coffee]: Do we want to return a string or JSON?

File: mock.coffee [hunk 2]
```diff
@@ -21,9 +30,22 @@ module.exports =
...
-        admin_sdk.data_orgunits_map[ou] = true for ou in admin_sdk.data.orgunits
+        admin_sdk.data_orgunits_map[ou] = generate_ou(ou) for ou in admin_sdk.data.orgunits
     admin_sdk.refresh_map()
 
+    # Currently, the mock library supports only this particular invocation of OrgUnitProvisioning.get
+    sandbox.stub admin_sdk.OrgUnitProvisioning.prototype, 'get', (customer_id, org_unit_path, cb) =>
+      # this method takes in org_unit_path without the leading slash, but our mock-db keys include the
+      # leading slash, so append it
+      existing_ou = admin_sdk.data_orgunits_map["/#{org_unit_path}"]
+      console.log _.keys admin_sdk.data_orgunits_map
```
Comment (alexandersoto) [inline mock.coffee]: debugging log?

File: test/google.coffee [hunk 3]
```diff
@@ -517,11 +523,25 @@ describe 'OrgUnitProvisioning', ()->
...
+      orgUnitId: 'ou_id1'
+      parentOrgUnitPath: '/'
+    second_ou =
+      kind: 'admin#directory#orgUnit'
+      name: 'Schoolname'
+      orgUnitPath: '/Students/SchoolName'
+      orgUnitId: 'ou_id2'
+      parentOrgUnitPath: '/Students'
+    cache =
+      '/Students': first_ou
```
Comment (alexandersoto) [inline test/google.coffee]: Nit, but one possible improvement is to use `first_ou.orgUnitPath`, fewer typo possibilities.

File: lib/org_unit_provisioning.coffee (hunk 1 above)
Comment (kshen0) [inline lib/org_unit_provisioning.coffee]: I believe `quest` parses the response to a javascript object.

File: mock.coffee (hunk 2 above)
Comment (kshen0) [inline mock.coffee]: 967cf57

File: test/google.coffee (hunk 3 above)
Comment (kshen0) [inline test/google.coffee]: 967cf57

**Comment by alexandersoto** [comment]
lgtm!

**Pull request merged**